File: frontend/src/services/addressService.js
import api from "./api";

export const getMyAddresses = async () => {
  const res = await api.get("/addresses");
  return res.data.data;
};

export const createAddress = async (data) => {
  const res = await api.post("/addresses", data);
  return res.data.data;
};

export const updateAddress = async (id, data) => {
  const res = await api.put(`/addresses/${id}`, data);
  return res.data.data;
};

export const deleteAddress = async (id) => {
  return api.delete(`/addresses/${id}`);
};

export const setDefaultAddress = async (id) => {
  const res = await api.patch(`/addresses/${id}/default`);
  return res.data.data;
};

File: backend/src/main/java/com/shop/features/address/service/AddressService.java
package com.shop.features.address.service;

import java.util.List;

import com.shop.features.address.dto.AddressRequestDto;
import com.shop.features.address.dto.AddressResponseDto;

public interface AddressService {

    AddressResponseDto createAddress(
            String email,
            AddressRequestDto request
    );

    List<AddressResponseDto> getMyAddresses(
            String email
    );
    AddressResponseDto updateAddress(
            String email,
            Long addressId,
            AddressRequestDto request
    );

    void deleteAddress(
            String email,
            Long addressId
    );

    AddressResponseDto setDefaultAddress(
            String email,
            Long addressId
    );
}